modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/ProcessInvoices.java

```diff
@@ -48,10 +49,22 @@ protected ActionResult action(JSONObject parameters, MutableBoolean isStopped) {
             log.debug(parameters.toString());
 
             for (Invoice invoice : input) {
-                var message = processInvoice(invoice, documentAction, voidDate, voidAcctDate);
+                Boolean lockedRecord=false;
+                //The following condition checks if the record is locked, this makes sense for Oracle databases
+                if (invoice.isProcessNow() && "CO".equals(documentAction)){
+                    lockedRecord=true;
+                }
+                //In case of a locked record, the docAction will be forced to XL, this will unlock the record and proceed to complete
+                var message = processInvoice(invoice,lockedRecord ? "XL": documentAction, voidDate, voidAcctDate);
+
                 if (message.getType().equals("Error")) {
                     errors++;
                 }
+                if (!message.getType().equals("Error") && lockedRecord){
```

Comment:
StringUtils instead string.equals, also you can improved this if condition evaluation using the evaluation of lockedRecord first as primitive boolean.